mobile/status.go

```diff
@@ -1023,6 +1023,21 @@ func GenerateImages(filepath string, aX, aY, bX, bY int) string {
 	return string(data)
 }
 
+// StartSearchForLocalPairingPeers starts a UDP multicast beacon that both listens for and broadcasts to LAN peers
+// on discovery the beacon will emit a signal with the details of the discovered peer.
+//
+// Currently, beacons are configured to search for 2 minutes pinging the network every 500 ms;
+//   - If no peer discovery is made before this time elapses the operation will terminate.
+//   - If a peer is discovered the pairing.PeerNotifier will terminate operation after 5 seconds, giving the peer
+//     reasonable time to discover this device.
+//
+// Peer details are represented by a json.Marshal peers.LocalPairingPeerHello
+func StartSearchForLocalPairingPeers() string {
```

Comment:
Thank you @ilmotta for your feedback, you make some good points.

> do you think clients might need to sometimes terminate the search before waiting for 2min? Just curious, would it be possible to implement something in the future to preemptively terminate the search? For example, a client might want or need to restart the search.

I actually struggled with this consideration myself, and I ultimately do not disagree with you. It is technically possible, the only route to achieving this would be to attach something linked to the peer search functionality to the `GethStatusBackend`. This is because the search functionality occurs in a sub routine which is inaccessible once fired, communication can only be done via `chan`. The lower level UDP functionality actually does use a "stop chan", which we use to terminate the search after a peer is discovered.

Attaching the stop `chan` to the backend would be the simplest way to allow for ad hoc termination, but it may pollute the concerns of the `GethStatusBackend`, see how all of the concerns of the of this struct are quite low level.

```go
// GethStatusBackend implements the Status.im service over go-ethereum
type GethStatusBackend struct {
	mu sync.Mutex
	// rootDataDir is the same for all networks.
	rootDataDir string
	appDB       *sql.DB
	config      *params.NodeConfig

	statusNode           *node.StatusNode
	personalAPI          *personal.PublicAPI
	multiaccountsDB      *multiaccounts.Database
	account              *multiaccounts.Account
	accountManager       *account.GethManager
	transactor           *transactions.Transactor
	connectionState      connection.State
	appState             appState
	selectedAccountKeyID string
	log                  log.Logger
	allowAllRPC          bool // used only for tests, disables api method restrictions

}
```

---

> Also, what if some client finds 2min too long or 500ms pings excessively short, and so on. I was wondering if it would be a good idea to change StartSearchForLocalPairingPeers to accept some arguments, but if they're not present allow Search to use sensible defaults (the values you already hardcoded seem good).

Yes we can make this configurable. I just pulled these numbers out of the air anyway 😆 .

Reply:
> Attaching the stop chan to the backend would be the simplest way to allow for ad hoc termination, but it may pollute the concerns of the GethStatusBackend

It's good to know it's technically viable without too much trouble. Since there's no clear need from clients nowadays, it's reasonable to me to delay implementing this capability. Thanks for the explanation and code example.

> Yes we can make this configurable

Nice! :bicyclist: